xarray/core/alignment.py

```diff
@@ -99,6 +105,21 @@ def align(
         obj, = objects
         return (obj.copy(deep=copy),)
 
+    if join == "override":
+        objects = list(objects)
+        new_indexes = dict()
+        for index, obj in enumerate(objects[1:]):
+            for dim in objects[0].indexes:
```

Comment:
There is a potential edge case here, where the first object does not have one of the dimensions found on multiple later objects, e.g.,
`align(DataArray(0), DataArray([1], [('x', [1])]), DataArray([2, 3], [('x', [2, 3])])`

This should be an error, but I think the current code would let this pass through, possibly creating an invalid object. Instead, I think the right strategy is to iterate everything, building up the set of indexes to match against as you go.

Another edge case that I think you might get wrong currently is something like this, where the last DataArray doesn't have one of the indexes at all:
`align(DataArray([1], [('x', [1])]), DataArray([2], [('x', [2])], DataArray(3))`

Reply:
Ah thanks. Fixed now. I've added these two examples as a test case.